app/controllers/api/v1/restaurants_controller.rb

```diff
@@ -1,9 +1,31 @@
 class Api::V1::RestaurantsController < ApplicationController
+  protect_from_forgery with: :null_session
+  protect_from_forgery unless: -> { request.format.json? }
+  before_action :authorize_user, except: [:index, :show]
+  
   def index
     render json: Restaurant.all
   end
 
   def show
-    render json: Restaurant.find(params[:id])
+    restaurant = Restaurant.find(params[:id])
+    user = current_user
+    response_body = []
+    response_body.push(user)
+    response_body.push(restaurant)
+    render json: response_body
```

Comment:
instead of pushing into an array, it might be clearer to create a hash
`render json: { restaurant: restaurant, user: user }`
that way when the response gets to the react component we can call on each key directly instead of its index